bmc-acf/acf_manager.cpp

```diff
@@ -0,0 +1,461 @@
+const std::string DBUS_INVENTORY_SYSTEM_OBJECT =
+    "/xyz/openbmc_project/inventory/system";
+const std::string DBUS_INVENTORY_ASSET_INTERFACE =
+    "xyz.openbmc_project.Inventory.Decorator.Asset";
+const std::string DBUS_SERIAL_NUM_PROP = "SerialNumber";
+const std::string UNSET_SERIAL_NUM_KEYWORD = "UNSET";
+const std::string DBUS_SOFTWARE_OBJECT = "/xyz/openbmc_project/software";
+const std::string DBUS_FIELDMODE_INTERFACE =
+    "xyz.openbmc_project.Control.FieldMode";
+const std::string DBUS_FIELD_MODE_PROP = "FieldModeEnabled";
+const int FAILURE = -1;
+
+bool readBinaryFile(const std::string fileNameParm,
+                    std::vector<uint8_t>& bufferParm)
+{
+    std::ifstream sInputFile;
+    if (!fileNameParm.empty())
+    {
+        sInputFile.open(fileNameParm.c_str(), std::ios::in | std::ios::binary);
+        if (sInputFile.is_open())
+        {
+            // Get the size of the file
+            sInputFile.seekg(0, std::ios::end);
+            std::streampos size = sInputFile.tellg();
+            sInputFile.seekg(0, std::ios::beg);
+
+            bufferParm.reserve(size);
+            bufferParm.assign(size, 0);
+
+            sInputFile.read((char*)bufferParm.data(), size);
+            sInputFile.close();
+
+            return true;
+        }
+        else
+        {
+            log<level::ERR>("Failed to open file ");
+        }
+    }
+    else
+    {
+        log<level::ERR>("filename empty");
+    }
+
+    return false;
+}
+
+std::string readMachineSerialNumberProperty(const std::string& obj,
+                                            const std::string& inf,
+                                            const std::string& prop)
+{
+    std::string propSerialNum = "";
+    std::string object = obj;
+    auto bus = sdbusplus::bus::new_default();
+    auto properties = bus.new_method_call(
+        "xyz.openbmc_project.Inventory.Manager", object.c_str(),
+        "org.freedesktop.DBus.Properties", "Get");
+    properties.append(inf);
+    properties.append(prop);
+    try
+    {
+        auto result = bus.call(properties);
+        if (!result.is_method_error())
+        {
+            std::variant<std::string> val;
+            result.read(val);
+            if (auto pVal = std::get_if<std::string>(&val))
+            {
+                propSerialNum.assign((pVal->data()), pVal->size());
+            }
+            else
+            {
+                log<level::ERR>("could not get the host's serial number\n");
+            }
+        }
+    }
+    catch (const std::exception& exc)
+    {
+        log<level::ERR>("dbus call for getting serial number failured: ");
+        propSerialNum = "";
+    }
+    return propSerialNum;
+}
+
+int readFieldModeProperty(const std::string& obj, const std::string& inf,
+                          const std::string& prop)
+{
+    bool propBool = false;
+    std::string object = obj;
+    auto bus = sdbusplus::bus::new_default();
```

Comment:
Improvement: dbus common function you can keep in utility files.. Refer the other repos , handling d-bus.